lib/components/RunFrameWithApi/EnhancedFileSelectorCombobox/EnhancedFileSelectorCombobox.tsx

```diff
@@ -100,7 +119,16 @@ export const EnhancedFileSelectorCombobox = ({
 
   useEffect(() => {
     setFile(currentFile)
-  }, [currentFile])
+    // Only auto-sync folder if user hasn't manually navigated and file exists
+    if (currentFile && !hasManuallyNavigated && files.includes(currentFile)) {
+      const fileDir = getDirectoryPath(currentFile)
+      if (fileDir !== "/") {
+        setCurrentFolder(fileDir)
+      } else {
+        setCurrentFolder("")
```

Comment:
never use empty strings!!! use null instead. Empty strings cause tons of problems